src/main/java/com/cloudbees/hudson/plugins/folder/config/AbstractFolderConfiguration.java

```diff
@@ -30,24 +32,33 @@ public static AbstractFolderConfiguration get() {
 
     @DataBoundConstructor
     public AbstractFolderConfiguration() {
-        load();
-        if (healthMetrics == null) {
+        this.load();
+    }
+
+    /**
+     * Auto-configure the default metrics afetr all plugins have been loaded.
+     */
+    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED, before = InitMilestone.JOB_LOADED)
```

Comment:
I don't remember the implications of having both `after` and `before` rules in the same initializer. But I guess you've verified it works as intended?

Reply:
I have tested this yes. I am not sure how to create a test for the initialization though ? 
The real fix is to perform the auto configuration outside the constructor and when plugins are loaded: `after = InitMilestone.EXTENSIONS_AUGMENTED`. I do believe that this is enough. I added the `before = InitMilestone.JOB_LOADED` because I feel this is consistent (the global metrics will be used on Folder's creation)